### ISSUE TITLE
wp_get_current_user() conflict with custom prefix & storage active

### ISSUE THREAD

**Issue by JulienMelissas** [Interfacelab/ilab-media-tools#23]
With a custom prefix defined, and storage activated, I get this error on storage activation, breaking the entire site (WSOD).

Based on [this thread](https://discourse.interfacelab.io/t/enabling-storage-results-in-white-screen-of-death/118), I've tracked it down to those lines.

According to [this post](https://stackoverflow.com/a/6127607/2757787), `wp_get_current_user` isn't available until the `plugins_loaded` or `init` hooks.

A current fix for the issue is to change [these lines](https://github.com/Interfacelab/ilab-media-tools/blob/master/classes/Utilities/Prefixer.php#L124-L128) like this:
```diff
-$user = wp_get_current_user();
+$user = 0;
+if(function_exists('wp_get_current_user')) {
+	$user = wp_get_current_user();
+}
$userName = '';
if($user->ID != 0) {
	$userName = sanitize_title($user->display_name);
}
```

But that doesn't actually solve the issue. I think whatever calls the prefixer (possibly the entire plugin) needs to be wrapped in `init` or `plugins_loaded`.

Related: https://discourse.interfacelab.io/t/enabling-storage-results-in-white-screen-of-death/118

**Comment by JulienMelissas** [comment]
For those reading, you can temporary bypass this with this branch, until we hear back on this issue: https://github.com/Craftpeak/ilab-media-tools/tree/user-check-fix

**Comment by Guibod** [comment]
Any news on this ?

**Comment by JulienMelissas** [comment]
I still haven't heard anything... we have been using our fork and checking back here periodically ¯\\_(ツ)_/¯

**Comment by jawngee** [comment]
Fixed in 86eff4e5957039a0079e8f433561d20ddffc8598

**Comment by jawngee** [comment]
The fix is to only actually hijack the `upload_dir` filter when you need to.  So you add the filter in the `wp_handle_upload_prefilter` filter and then remove it in the `wp_handle_upload` filter.
